enable vm-image-builder to build Arm64 images (#732)
kubevirt/kubevirtci: cluster-provision/images/vm-image-builder/README.md

- qemu-img
- virt-install

> To do cross build for Arm64 image on AMD64 machine, following RPM needed:

Reviewer: ```suggestion
To cross build the Arm64 image on AMD64 machines,  the following RPM needs to installed:
```

publish-containerdisk.sh example localhost:1234/myimage:mytag
```

> To build `example` for arm64(aarch64), only need to set the environmental

Reviewer: ```suggestion
To build for arm64(aarch64), you need to set the following environmental
```

Reviewer 2: If we're touching the documentation, I suggest also replacing all instances of "environmental variable" with "environment variable", which AFAIK is the most commonly used spelling.

Reviewer: I read it as `enviroment` - tricked my own brain - thanks for catching that. 

You're right I'll update the suggestion.

variable.
```
export ARCHITECTURE=arm64
```

> To build the Virtual Machine without console output, only need to set the
> environmental variable CONSOLE. This is useful when the building script
> run in a CICD pipeline.

Reviewer: ```suggestion
To build the Virtual Machine without console output, you need to set the
environmental variable `CONSOLE`. This is useful when the build script is
run in a CICD pipeline.
```

```

> export WITHCONSOLE=no

Reviewer 2: This documents `WITHCONSOLE` but AFAICT the actual variable looked at is `CONSOLE`.

```
Then follow the previous step.
## Create a new containerdisk

Every directory contains build instructions for virt-customize. The build
instructions are distributed between the following three files:

```
$ ls -l example/
cloud-config		# cloud-init configuration for virt-customize
image-url		# download URL of the base image
image-url-arm64		# download URL of the base image for Arm64

> os-variant		# operating system variant (for example fedora32)

Reviewer 2: These are still not aligned O:-) Can you please just replace the tabs with spaces? That should solve it once and for all :)

$ kubectl wait --for=condition=AgentConnected vmi $VMI_NAME --timeout 5m
$ virtctl console testvm1
```

### Build and publish multi-arch images

> # It does not support to build alpine-cloud-init, because the project
> # https://raw.githubusercontent.com/alpinelinux/alpine-make-vm-image/master/alpine-make-vm-image
> # does not support build Arm64 image.

Reviewer: ```suggestion
The multi-arch publish does not support building alpine-cloud-init because the [alpine-make-vm-image](https://raw.githubusercontent.com/alpinelinux/alpine-make-vm-image/master/alpine-make-vm-image) project does not support building Arm64 images.
```

The `publish-multiarch-containerdisk.sh` script now supports building Arm64 and AMD64 images.
The script primarily performs the following tasks:

> 1. According to OPTIONS to execute specific actions.
> 2. Use `create-containerdisk.sh` to build images.
> 3. Upload the resulting images to a specific registry.
> 4. Upload multi-arch image manifest.

Reviewer: I am not sure if the first point adds anything here. 

```suggestion
1. Use `create-containerdisk.sh` to build images.
2. Upload the resulting images to a specific registry.
3. Upload multi-arch image manifest.
```

```
./publish-multiarch-containerdisk.sh -h
    Usage:
        ./publish_multiarch_image.sh [OPTIONS] BUILD_TARGET REGISTRY REGISTRY_ORG

    Build and publish multiarch infra images.

    OPTIONS
        -n  (native build) Only build image for host CPU Arch.
        -h  Show this help message and exit.
        -b  Only build the image and exit. Do not publish the built image.

# build and push multi-arch example image
./ publish-multiarch-containerdisk.sh example myregistry registry_org

# The script will do following things:
# 1. Build both Arm64 and AMD64 example image.
# 2. Generate a tag based on the current time.
# 3. Push the registry_org/myregistry/example:tag-arm64 and registry_org/myregistry/example:tag-amd64.
# 4. Generate a multi-arch manifest for the image, registry_org/myregistry/example:tag.
```

> To use your own tag for the image, only need to set the environmental variable.

Reviewer: ```suggestion
To use your own tag for the image, you need to set the environmental variable `KUBEVIRTCI_TAG`.
```
